config.py

```diff
@@ -20,6 +21,17 @@ class ConfigTemplater(object):
     def add_template(self, template):
         self.t[template.name] = template
 
+    def additional_default_options(self):
+        output = ''
+        base_options = ['redispatch', 'http-server-close', 'dontlognull']
+        if 'HAPROXY_DEFAULT_OPTIONS' in os.environ:
```

Comment:
Another nit: maybe instead of `HAPROXY_DEFAULT_OPTIONS` we could call it `HAPROXY_GLOAL_DEFAULT_OPTIONS`.